internal/compiler/object_tree.rs

```diff
@@ -243,6 +243,10 @@ impl Document {
         self.exports.iter().filter_map(|e| e.1.as_ref().left()).filter(|c| !c.is_global()).cloned()
     }
 
+    pub fn exposed_structs_and_enums(&self) -> Vec<Type> {
+        self.used_types.borrow().structs_and_enums.clone()
```

Comment:
This is not only the "exposed" structs and enum, this is all of the used one.

What you want really is ` self.exports.iter().filter_map(|e| e.1.as_ref().right())`
(Same as the function above, but right instead of left)

Reply:
Hmmm, I don't think I want just `exports`. Or at least, I want to mirror what we do for Rust and C++, and that's this:

```rust
    let (structs_and_enums_ids, structs_and_enum_def): (Vec<_>, Vec<_>) = doc
        .used_types
        .borrow()
        .structs_and_enums
        .iter()
        .filter_map(|ty| match ty {
            Type::Struct { fields, name: Some(name), node: Some(_), rust_attributes } => {
                Some((ident(name), generate_struct(name, fields, rust_attributes)))
            }
            Type::Enumeration(en) => Some((ident(&en.name), generate_enum(en))),
            _ => None,
        })
        .unzip();
```

I guess that's because this also makes implicitly used structs accessible, such as `Date` or `Time` if you use the `DatePicker`.

I'll remove the function then, as indeed `used_types`, etc. are accessible.

Reply:
C++ and Rust generates all used structs and enum because they are used in the generated code. Ideally, only the exported ones should be made available outside of the internal mod/namespace. 

(But we still have issues when two non-exported struct have the same name. (eg, we had the problem internally because a `TextStyle` structure was used both in a demo, and in one of the style, causing an error, despite none of them was exported to the final app)